src/OldPartitionsRemover/BySpaceRemoving/Arguments.cs

```
@@ -17,7 +17,14 @@ public class Arguments : CommonArguments
         public int DelayMilliseconds { get; set; }
 
         [Option("threshold-type", Default = "free", HelpText = "Type of threshold: `free` space on node or bob's `occupied` space")]
-        public string ThresholdType { get; set; } // Enums are case sensitive in CommandLineParser by default, and changing this requires recreating whole help
+        public string ThresholdTypeString { get; set; } // Enums are case sensitive in CommandLineParser by default, and changing this requires recreating whole help
+
+        public Result<ThresholdType> GetThresholdType()
+        {
+            if (Enum.TryParse<ThresholdType>(ThresholdTypeString, true, out var tt))
+                return Result<ThresholdType>.Ok(tt);
+            return Result<ThresholdType>.Error("Wrong threshold type provided");
```

This is an error due to incorrect user input. Obviously, the program should not try to do anything in this case, so IMHO it is better to throw an exception here

---

It's the same way threshold error is returned, both will be printed as errors without the need for any exceptions

---

Ok